src/Transform/ONNX/ConstProp.cpp

```diff
@@ -289,12 +289,62 @@ DenseElementsAttr ConstPropElementwiseUnary(
   auto rank = denseAttr.getType().getShape().size();
   SmallVector<uint64_t, 4> indices(rank, 0);
   std::vector<Attribute> resVector;
-  RecurseConstProppElementwiseUnary<ElementwiseUnaryOp>(
+  RecurseConstPropElementwiseUnary<ElementwiseUnaryOp>(
       rewriter, resVector, denseAttr, indices, rank);
   ArrayRef<Attribute> resRef(resVector);
   return DenseElementsAttr::get(resType, resRef);
 }
 
+//===----------------------------------------------------------------------===//
+// Code to perform constant propagation for transpose.
+//===----------------------------------------------------------------------===//
+
+void RecurseConstPropTranspose(PatternRewriter &rewriter,
+    std::vector<Attribute> &resVector, DenseElementsAttr &attr,
+    SmallVector<uint64_t, 4> &indices, SmallVector<uint64_t, 4> &perm,
+    int freeRank) {
+  // printf("recurse with free %d\n", freeRank);
```

Comment:
Are we sure about checking in this line of code?

Reply:
tx